include/rangers-api/ApplicationCommon/FSM/GOCTinyFsm2.h

```diff
@@ -0,0 +1,96 @@
+#pragma once
+
+namespace app_cmn::fsm {
+    template<typename T>
+    class GOCTinyFsm2;
+
+    template<>
+    class GOCTinyFsm2<hh::game::GameObject> : public hh::game::GOComponent {
+    public:
+        struct Event : public hh::ut::TinyFsmEvent {
+            hh::fnd::Message* message;
+            void* updateEvent;
+        };
+        typedef hh::ut::TinyFsm<hh::game::GameObject, Event> Fsm_t;
+        typedef typename Fsm_t::State_t State_t;
+        typedef typename Fsm_t::Event_t Event_t;
+        struct alignas(8) Config {
+            State_t initialState;
+            bool setFlag3;
+            hh::game::GameObject* targetObject;
+            uint32_t unk1;
+            uint32_t unk2;
+            bool setGOCFlag0;
+            bool setGOCFlag1;
+            bool setGOCFlag2;
+            bool setFlag4;
```

Comment:
Variable namings should be done in title case.
Pointer types should be identified by a `p` before the variable name.